src/cmd/ksh93/tests/io.sh

```diff
@@ -588,3 +588,11 @@ n=$( exec {n}< /dev/null; print -r -- $n)
 
 # According to POSIX <> should redirect stdin when no fd is specified
 read -n 1 -t 0.1 <>/dev/zero || log_error '<> should redirect stdin by default'
+
+TMPF=$(mktemp -p $TEST_DIR)
```

Comment:
Don't use `mktemp -p` as it isn't available on BSD/macOS. Since the test harness has already created a unique test dir and this test doesn't require a unique file you can just hardcode the file name.